### PR TITLE
Addressing issue #2283 where getHost is throwing NPE

### PR DESCRIPTION
**Pull request by ctoestreich**
Fixing NPE from get host and using host OR authority.

### CODE REVIEW

File: http/src/test/groovy/io/micronaut/http/HttpRequestSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+package io.micronaut.http
+
+import spock.lang.Specification
+
+class HttpRequestSpec extends Specification {
+
+    def "Converting to url and getting host and authority work as expected"() {
+        expect:
+        verifyAll {
```
Comment (jameskleeh) [inline http/src/test/groovy/io/micronaut/http/HttpRequestSpec.groovy]: What is the purpose of `verifyAll` with asserts? Couldn't it just be `expect:` with each `HttpRequest... ==` on its own line?

**Comment by ctoestreich** [comment]
@jameskleeh review cleanup complete

**Comment by jameskleeh** [comment]
@ctoestreich We don't really have any specifics. I requested the revert because you didn't add any imports so no change was not necessary. We generally don't want commits for features/fixes to contain non essential changes like whitespace, etc.

**Comment by jameskleeh** [comment]
I implemented this differently because I wanted to not repeat the logic and also use it in other places in the client. The changes went into 1.2.6

**Pull request closed without merging**